docs/GenericChart.md

```diff
@@ -0,0 +1,428 @@
+potentially customise filters for other applications of GenericChart e.g. some filter options might not be available 
+for some charts. 
+- `Plotly`: an implementation which combines provided chart data and metadata as a JSONata template to a full plotly 
+configuration and loads it using the Plotly library. This is very similar to the Chart component in comet. Minor changes 
+from that implementation are that we are using the `newPlot` method instead of `react` to accommodate updates to chart 
+height when the number of subplots change due to filter changes (particularly are level), and we are using a more 
+lightweight Plotly distribution, plotly-js.basic-dist.
+
+This is how they look when rendered in HINT to display the Input Time Series chart:
+![Generic Chart Components](GenericChartComponents.png "Generic Chart Components")
+
+### Generic Chart Metadata
+
+The Generic Chart Metadata consists of a dictionary of `GenericChartMetadata` objects, whose keys are the chart ids. 
+Here is an annotated example of GenericChartMetadata:
+```
+{
+      // This section configures the datasets available to the charts, and where to get them from
+      "datasets": [  
+        {
+          // id by which chart will identify dataset - the id used in the dict of datasets in the store
+          "id": "art",  
+
+          // translation key of label to display for the option for this dataset in the Data source component
+          "label": "ART",  
+
+          // endpoint from which to fetch this dataset
+          "url": "/chart-data/input-time-series/programme",
+
+	      // which filters to display for this dataset, and where to get their metadata (i.e. their filter options). 
+	      // "data" is the only currently supported option i.e. get the filters from the metadata property dataset itself	
+          "filters": [ 
+            {
+              "id": "plot_type",
+              "source": "data"
+            },
+            {
+              "id": "area_level",
+              "source": "data"
+            },
+            {
+              "id": "time_step",
+              "source": "data"
+            }
+          ]
+        },
+        {
+          "id": "anc",
+          "label": "ANC",
+          "url": "/chart-data/input-time-series/anc",
+          "filters": [
+            {
+              "id": "plot_type",
+              "source": "data"
+            },
+            {
+              "id": "area_level",
+              "source": "data"
+            },
+            {
+              "id": "age",
+              "source": "data"
+            }
+          ]
+        }
+      ],
+      "dataSelectors": {
+	// Data source is currently the only type of data selector supported.
+	// A data source has a particular role in the chart. The user selects which dataset to use for each data source. 
+        "dataSources": [
+          {
+            // Id for this data source
+            "id": "data",
```

Comment:
What is the purpose of this ID?

Also, I think it's not ideal to use `data` as an example of an ID because a) it's a data source, not data itself, right? and b) when I see `"source": "data"` above, and a data source with an ID of `"data"` here I automatically link the two in my head, whereas they're not related, are they?

Reply:
> The id is just for distinguishing data sources from each other.

And how/where is it used? (Sorry!)

Reply:
It is used in the component to keep track of the selections for each data source (its selected dataset, and selected filter options). It's also used to structure the data passed to plotly, and hence is referenced in the plotly jsonata. So when the jsonata uses e.g. `data.area_name` that 'data' is derived from this id. 
(So actually I maybe won't change this id as that would impact the jsonata, but I'll try to explain it better in the comment.)

Reply:
Ok, cool. I guessed it might be used in front-end, but not in jsonata, so that's definitely worth commenting on IMO. Thanks!